lib/fund.js

```diff
@@ -37,161 +32,174 @@ fundCmd.completion = function (opts, cb) {
+  const result = {}
+  const seenUrls = new Map()
+  let prev = fundingInfo
+  let parent = result
+
+  const tree = obj =>
+    archy(obj, '', { unicode: opts.unicode })
+
+  breadth({
+    tree: prev,
+    visit: ({ name, version, funding }) => {
+      const { url } = funding || {}
+      const pkgRef = getPrintableName({ name, version })
+      const label = url ? tree({
+        label: url,
+        nodes: [pkgRef]
+      }).trim() : pkgRef
+      let item = {
+        label,
         nodes: []
       }
 
-      if (printableType) {
-        result.nodes.push(printableType)
-      }
-
-      if (printableUrl) {
-        result.nodes.push(printableUrl)
+      if (seenUrls.has(url)) {
+        item = seenUrls.get(url)
+        item.label += `, ${pkgRef}`
+      } else {
+        parent.nodes = (parent.nodes || []).concat(item)
+        seenUrls.set(url, item)
       }
 
-      setStackedItem(funding, result)
-
-      return result
-    }).reduce((res, result) => {
-      // recurse and exclude nodes that are going to be stacked together
-      const { fundingItem } = result
-      const { dependencies, funding } = fundingItem
-      const items = getFundingItems(dependencies)
-      const stackedResult = retrieveStackedItem(funding)
-      items.forEach(i => result.nodes.push(i))
-
-      if (stackedResult && stackedResult !== result) {
-        stackedResult.label += `, ${result.label}`
-        items.forEach(i => stackedResult.nodes.push(i))
-        return res
+      return item
+    },
+    getChildren: (node, nodeResult) => {
+      if (prev.dependencies &&
+        !Object.hasOwnProperty.call(prev.dependencies, node.name)) {
+        prev = node
+        parent = nodeResult
       }
-
-      res.push(result)
-
-      return res
-    }, [])
-
-  const [ result ] = getFundingItems({
-    [fundingInfo.name]: {
-      dependencies: fundingInfo.dependencies,
-      funding: fundingInfo.funding,
-      version: fundingInfo.version
+      return Object.keys(node.dependencies || {})
+        .map(key => ({
+          name: key,
+          ...node.dependencies[key]
+        }))
     }
   })
 
-  return archy(result, '', { unicode: opts.unicode })
+  const [root] = result.nodes
+  return tree(root.label !== header ? { label: header, nodes: [root] } : root)
 }
 
-function openFundingUrl (packageName, cb) {
-  function getUrlAndOpen (packageMetadata) {
-    const { funding } = packageMetadata
-    const { type, url } = retrieveFunding(funding) || {}
-    const noFundingError =
-      new Error(`No funding method available for: ${packageName}`)
-    noFundingError.code = 'ENOFUND'
-    const typePrefix = type ? `${type} funding` : 'Funding'
-    const msg = `${typePrefix} available at the following URL`
-
-    if (validFundingUrl(funding)) {
-      openUrl(url, msg, cb)
+function openFundingUrl ({ actualTree, spec, fundingSourceNumber }) {
+  const retrievePackageMetadata = () => {
+    const arg = npa(spec, actualTree.path)
```

Comment:
I'm a little confused by this bit.  If you have a nested dep, like:

```
root
+-- x
    +-- y
```

And the dependency from `x` to `y` is `file:./y`, then won't this end up with a npa spec resolving to `root/y` instead of `root/node_modules/x/y`?